### ISSUE TITLE
Unable to locate element using Accessibility ID, iOSNsPredicate and iOSClassChain

### ISSUE THREAD

**Issue by vpkhachane** [appium/appium#15099]
Hello Team,

I am unable to locate mobile elements using Accessibility ID, iOSNsPredicate and iOSClassChain in my iOS automation framework . Following is my code structure

@iOSXCUITFindBy(accessibility = "loginKashtagInput")
   public static MobileElement kashtag;

But i am getting following error

[WD Proxy] Got response with status 404: {"value":{"error":"no such element","message":"unable to find an element using 'accessibility id', value 'loginKashtagInput'","traceback":"(\n\t0   WebDriverAgentLib                   0x000000010b7660ca FBNoSuchElementErrorResponseForRequest + 298\n\t1   WebDriverAgentLib                   0x000000010b765e89 +[FBFindElementCommands handleFindElement:] + 409\n\t2   WebDriverAgentLib                   0x000000010b721856 -[FBRoute_TargetAction mountRequest:intoResponse:] + 182\n\t3   WebDriverAgentLib                   0x000000010b6fca88 __37-[FBWebServer registerRouteHandlers:]_block_invoke + 536\n\t4   WebDriverAgentLib                   0x000000010b748a2b -[RoutingHTTPServer handleRoute:withRequest:response:] + 219\n\t5   WebDriverAgentLib                   0x000000010b749923 __72-[RoutingHTTPServer routeMethod:withPath:parameters:request:connection:]_block_invoke + 83\n\t6   libdispatch.dylib                   0x00007fff20106508 _dispatch_client_callout + 8\n\t7   libdispatch.dylib                   0x000...

Then i tried iOSNsPredicate and iOSClassChain but getting same error
@iOSXCUITFindBy(iOSNsPredicate = "name == loginKashtagInput")
@iOSXCUITFindBy(iOSClassChain = "**/XCUIElementTypeTextField[`name == \"loginKashtagInput\"`]")
 

## Environment

* Appium version 1.20.0
* Desktop OS/version used to run Appium: macOS Big Sur 11.2
* Node.js version (unless using Appium.app|exe): v14.2.0
* Npm or Yarn package manager: npm version 6.14.4
* Mobile platform/version under test: iPhone 12 pro Max 14.3 simulator
* Real device or emulator/simulator: Simulator
* Appium CLI or Appium.app|exe: Started Appium through Java Code

Appium Log

<img width="1280" alt="Screenshot 2021-02-10 at 5 17 54 PM" src="https://user-images.githubusercontent.com/15686482/107506690-7542b480-6bc4-11eb-83f9-0e435f5b156b.png">

**Comment by mykola-mokhnach** [comment]
Is the element present in the page source?

**Comment by vpkhachane** [comment]
@mykola-mokhnach  Yes Elements are present on the screen
![Uploading Screenshot 2021-02-10 at 5.26.41 PM.png…]()

**Comment by mykola-mokhnach** [comment]
It does not usually work this way. It is necessary to take the page source right before locating the element to make sure it is there.

**Comment by vpkhachane** [comment]
@mykola-mokhnach  I am able access element using xpath . Only getting issue while using accessibility id

**Comment by mykola-mokhnach** [comment]
this could be a race condition. Usually xpath locators are slow, so the app has more time to update the UI, which is not the case for native locators. Try to apply an explicit wait in order to locate the element

**Comment by vpkhachane** [comment]
@mykola-mokhnach I already added explicit wait in order to. locate element . Please find below code snippet . let me know if any changes necessory in it. 
 
WebDriverWait waitForkashtagtextbox = new WebDriverWait(Test_Env_Setup_AppiumController.driver, 30);
         waitForkashtagtextbox.until(ExpectedConditions.elementToBeClickable(By.xpath("//XCUIElementTypeTextField[@name=\"kashtagInput\"]")));
kashtag.clear();
kashtag.sendKeys(username);

**Comment by mykola-mokhnach** [comment]
Sorry, I don't get what the issue is. The code above is still using xpath strategy instead of accessibility id

**Comment by vpkhachane** [comment]
@mykola-mokhnach  Sorry Its my mistake. Now i have updated code . Still i am getting same issue.

 WebDriverWait waitForkashtagtextbox = new WebDriverWait(Test_Env_Setup_AppiumController.driver, 30);
           waitForkashtagtextbox.until(ExpectedConditions.elementToBeClickable(MobileBy.AccessibilityId("loginKashtagInput")));
kashtag.clear();
kashtag.sendKeys(username);

**Comment by mykola-mokhnach** [comment]
Try to enable Xcodebuild and device logging. Usually XCTest is pretty talkative about what it is doing there.
Also you could try to print the native page source, because this is the tree where XCTest performs native lookups. Use `mobile: source` endpoint with format set to `description` for such purpose: https://github.com/appium/appium-xcuitest-driver#mobile-source

**Comment by vpkhachane** [comment]
@mykola-mokhnach  I am locate textmessage using accessibility ID . Only getting using while accessing textboxes . Can you please help me on this confusion.  Why am not able to access username and password textboxes.

Following element working fine using accessibility ID
@iOSXCUITFindBy(accessibility = "Choose the method to login")
public static MobileElement loginScreenTxtValMsg;

**Comment by mykola-mokhnach** [comment]
Like I said above there is nothing I could do without the actual XCTest log since the location and the hard lifting there is done by XCTest and we are limited by its API usage. Could also be related to https://github.com/appium/appium/issues/15089

**Comment by mykola-mokhnach** [comment]
Closed because of no response
